### ISSUE TITLE
Instead of auto-fetching child-manifests, use recommendation system

### ISSUE THREAD

**Issue by ben-edna** [dfetch-org/dfetch#242]
Currently child projects can have their own manifest file. This is automatically also processed.
However this can lead to multiple project requiring the same dependency and multiple fetches of that dependency.
Also the sub project determines the location of the dependency.

A better approach would be to:

- Update main manifest (`dfetch update`)
- Find all child manifests
- Create a single set of required projects as listed by the fetched projects
- Check if any of the listed projects are already part of the main manifest
- If any project is missing, print a message:
```
Some-subproject requires "CppUtest" to function, please add following to your manifest:

  - name: CppUTest
     url: https://www.github.com/cpputest/cpputest
     tag: v3.2.0
``` 

This system could also warn if there are version clashes

**Comment by ben-edna** [comment]
@jgeudens @sach-edna @KrisEDNA : 

I've created a proof of concept in #262: In this DFetch outputs the following if a child-manifest has a dependency that is not part of the main manifest:

```console
Dfetch (0.5.1)
Multiple manifests found, using dfetch.yaml 
  SomeProject         : up-to-date (v1)

SomeProject depends on the following project(s) which are not part of your manifest:

-   name: SomeOtherProject
    url: some-remote-server/SomeOtherProject.git
    tag: v1
```

For more examples see: [check-child-projects.feature](https://github.com/dfetch-org/dfetch/blob/feature/recommend-children/features/check-child-projects.feature)

**Comment by jgeudens** [comment]
I like this approach because it lets the user handle all the required dependencies. The tool doesn't force a particular way to solve it.

Small improvement would be to show a bit more detail of the path to the used manifest.

**Comment by spoorcc** [comment]
Thanks! Good idea about showing more details :+1:

**Comment by ben-edna** [comment]
Added the source file that created the recommendation and also added the recommendation system during updates:

```console
Dfetch (0.5.1)
  SomeProjectWithChild: Fetched v1

"SomeProjectWithChild" depends on the following project(s) which are not part of your manifest:
(found in ThirdParty/SomeProjectWithChild/dfetch.yaml)

-   name: SomeOtherProject
    url: some-remote-server/SomeOtherProject.git
    tag: v1

  SomeProjectWithoutChild: Fetched v1
```

**Comment by ben-edna** [comment]
Fixed in #262
